eclair-core/src/main/scala/fr/acinq/eclair/wire/internal/channel/version3/ChannelCodecs3.scala

```diff
@@ -255,7 +257,7 @@ private[channel] object ChannelCodecs3 {
 
     val spentMapCodec: Codec[Map[OutPoint, Transaction]] = mapCodec(outPointCodec, txCodec)
 
-    val commitmentsCodec: Codec[Commitments] = (
+    val readOnlyCompatibilityCommitmentsCodec: Codec[Commitments] = (
```

Comment:
I think the change is big enough that it's worth creating a `ChannelCodecs4` for the new codecs, which will make the code changes to that file smaller and easier to reason about.

As discussed offline, we should also "freeze" the `master` branch while we're converging on a data model that works for splicing: while `master` is locked, we can make any breaking changes that we want to the new `ChannelCodecs4`, which will make our lives easier (for reviews and actual development).

Reply:
Did the `ChannelCodecs4` in [64e1f21](https://github.com/ACINQ/eclair/pull/2540/commits/64e1f218af151ea9afba08a643e3f5de7d151e46).